orbit/pkg/update/nudge_test.go

```diff
@@ -56,10 +59,24 @@ func (s *nudgeTestSuite) TestNudgeConfigFetcherAddNudge() {
 		opt:    Options{Targets: make(map[string]TargetInfo), RootDirectory: tmpDir},
 	}
 	runner := &Runner{updater: updater, localHashes: make(map[string][]byte)}
-	interval := time.Minute
+	interval := time.Second
 	cfg := &fleet.OrbitConfig{}
 	nudgePath := "nudge/macos/stable/nudge.app.tar.gz"
+
+	// set up mock runNudgeFn to capture exec command
+	var execCmd func(command string, args ...string) *exec.Cmd
+	var execOut string
+	runNudgeFnInvoked := false
 	runNudgeFn := func(execPath, configPath string) error {
+		runNudgeFnInvoked = true
+		if execCmd != nil {
+			cmd := execCmd(execPath, configPath)
+			out, err := cmd.Output()
+			if err != nil {
+				return err
+			}
+			execOut = string(out)
```

Comment:
Personally, this approach gave me more confidence in terms of completeness and understanding what's happening in practice rather than just trying to instantiate `exec.ExitError` directly (which in turn depends on an instance of `os.ProcessState` and so on and so on). 

FWIW, the approach follows the pattern used by the `exec` package [itself](https://github.com/golang/go/blob/master/src/os/exec/exec_test.go#L136).

And this is set up so it's still easy to mock the `runNudgeFn` without using `execCmd` for those cases where we're less concerned with testing the os exit event itself.

Reply:
to me, the clear advantage of returning a mocked value here is that the test is easy to understand at first glance.

I was able to quickly understand what's going on here just because I was familiar with this trick (I found myself in the same position as you not so long ago), but it's not super simple to follow.

>  trying to instantiate exec.ExitError directly (which in turn depends on an instance of os.ProcessState and so on and so on).

curious, did you find the need to mock a function that was deep down the chain? can we get away with just implementing what we need?

Reply:
> curious, did you find the need to mock a function that was deep down the chain? can we get away with just implementing what we need?

I didn't get too deep into it. I found that I'd need to at least mock some of the `ProcessState` because  it gets used in `ExitError.Error` [here](https://cs.opensource.google/go/go/+/refs/tags/go1.20.6:src/os/exec/exec.go;l=861), which in turn calls a custom stringer [here](https://cs.opensource.google/go/go/+/refs/tags/go1.20.6:src/os/exec_posix.go;l=98). At that point, I zoomed out and looked to how others have approached the problem. 

I'm sure that I could mock it out if necessary, but I appreciated that this approach let me see it in action as a separate process. And honestly, I wanted to leave behind some artifacts of this approach in case we need to use it again in the future.

Reply:
>  And honestly, I wanted to leave behind some artifacts of this approach in case we need to use it again in the future.

I see that as a downside, but we can agree to disagree! will approve regardless of the approach here.

Reply:
I'm all for a more streamlined approach. This was just the best one that I found. 

The challenge is the things that need to be mocked aren't typed as interfaces in a way that make it easy to swap out. For example, I don't see a good way to mock around this [line](https://cs.opensource.google/go/go/+/refs/tags/go1.20.6:src/os/exec_posix.go;l=102).